webapp-django/crashstats/crashstats/static/crashstats/js/socorro/oauth2.js

```diff
@@ -110,8 +121,18 @@ var OAuth2 = (function() {
                         $.post(url, data)
                         .done(function(response) {
                             // It worked!
-                            // TODO: https://bugzilla.mozilla.org/show_bug.cgi?id=1283296
-                            document.location.reload();
+                            var next = getParameterByName('next');
+                            // only if ?next=/... exists on the current URL
+                            if (next) {
+                                // A specific URL exits.
+                                // This is most likely the case when you tried
+                                // to access a privileged URL whilst being
+                                // anonymous and being redirected.
+                                // Make sure it's on this server
+                                document.location.pathname = next;
+                            } else {
```

Comment:
@peterbe: I don't really understand what you tried to do. I was thinking you'd create a new `URL` object using the next value and then set the pathname. Something like this:

```
var thisURL = URL(document.location.href);
var newURL = thisURL.searchParams['next'];

if (next) {
    documentation.location.pathname = newURL.pathname;
} else {
    ....
}
```

Why wouldn't that work here?